persistence-modules/spring-data-jpa-repo-3/src/main/java/com/baeldung/spring/data/jpa/naturalid/entity/ConferenceRoom.java

```diff
@@ -0,0 +1,56 @@
+package com.baeldung.spring.data.jpa.naturalid.entity;
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import org.hibernate.annotations.NaturalId;
+
+@Entity
+public class ConferenceRoom {
+	@Id
+	@GeneratedValue(strategy = GenerationType.IDENTITY)
+	private Long id;
+
+	@NaturalId(mutable = true)
+	private String name;
+
+	private int capacity;
+
+	public ConferenceRoom(String name, int capacity) {
+		this.name = name;
+		this.capacity = capacity;
+	}
+
+	protected ConferenceRoom() {
+	}
+
+	public Long getId() {
+		return id;
+	}
+
+	public void setId(Long id) {
```

Comment:
Why have a setter for id but not for name, which you marked as mutable? Seems inconsistent. The other entity also does not have a setter for id.

Reply:
removed the `(mutable = true)`. It was something I was experimenting with and forgot to remove afterward. Additionally, I have removed the setter for the id as well.

Reply:
Agree.